gen_png.py

```diff
@@ -74,15 +74,28 @@ def git_commit_raster():
     print_pipe(stream.read())
 
 # create a new tag
-def create_tag():
+def create_tag(added, modified, deleted, renamed_old, renamed_new):
     """Creates a new git tag and names it appropriately"""
     # get tags
     stream = os.popen('git tag -l "auto-v*"')
     output = stream.read()
     tags = output.split('\n')
     tags.remove('')
+    # create tag message
+    added = '\n'.join([x.removeprefix('svg/').removesuffix('.svg') for x in added])
+    modified = '\n'.join([x.removeprefix('svg/').removesuffix('.svg') for x in modified])
+    deleted = '\n'.join([x.removeprefix('svg/').removesuffix('.svg') for x in deleted])
+    
+    i = 0
+    renamed = ''
+    while i < len(renamed_old):
```

Comment:
How do you ensure that someone doesn't accidentally call this function where `renamed_old` has a different length than `renamed_new`?

Reply:
renames can only occur in pairs 
so that would be a usage error of the method not a fault of the method itself
I can catch that case and fail, but I think the back trace from it exploding is fine imo

Reply:
IMO a simple check on whether the length for both are the same and if they aren't throw, would be better.
As not checking could lead to harder-to-debug usage errors (i.e. i accidentally put in renamed_old with length 1, but renamed_new with 10 for reasons:tm:).